### PR TITLE
docs(semver): improve docs

### PR DESCRIPTION
**Pull request by kt3k**
part of #3764 

This improves the API docs of `@std/semver`

### CODE REVIEW

File: semver/can_parse.ts [hunk 1]
```diff
@@ -2,6 +2,21 @@
 // This module is browser compatible.
 import { parse } from "./parse.ts";
 
+/**
+ * Returns true if the string can be parsed as semver.
+ *
+ * @example Usage
+ * ```ts
+ * import { canParse } from "@std/semver";
```
Comment (magurotuna) [inline semver/can_parse.ts:10]: I'm wondering if there is a general guideline that we can follow on whether we should put module-level import (i.e. `from "@std/semver"`) or one deeper one (i.e. `from "@std/semver/can-parse"`). Do you have any thought?

Comment (kt3k) [reply semver/can_parse.ts:10]: Currently we don't have explicit policy about it

Comment (iuioiua) [reply semver/can_parse.ts:10]: We should use the most specific import specifier. It encourages keeping dependencies leaner and aligns with JSR.

File: semver/range_max.ts [hunk 2]
```diff
@@ -44,15 +44,21 @@ function comparatorMax(comparator: Comparator): SemVer {
 }
 
 /**
+ * The maximum valid SemVer for a given range or INVALID
+ *
+ * @example Usage
+ * ```ts
+ * // This API is deprecated
```
Comment (iuioiua) [inline semver/range_max.ts:51]: Perhaps, we can have two snippets in this example; one for the deprecated usage and one for the new API's usage, like a migration guide.

**Review by iuioiua — CHANGES_REQUESTED** [review]
Great to have this done! My main nit is the proper casing of "SemVer".

**Review by iuioiua — APPROVED** [review]
Nice!

**Pull request merged**